mindsdb/integrations/handlers/autokeras_handler/autokeras_handler.py

```diff
@@ -0,0 +1,158 @@
+    """Helper function to trains an AutoKeras model with an input df.
+
+    Automatically decides on classification vs. regression depending on
+    the type of the target.
+
+    Will auto-encode categorical variables as dummies.
+
+    Returns both trained model and the names of categoric dummy columns, which
+    are passed later to the prediction method.
+    """
+    # Choose regressor of classifier based on target data type
+    if np.issubdtype(df[target].dtype, np.number):
+        mode = "regression"
+        y_train = df[target]
+    else:
+        mode = "classification"
+        lb = preprocessing.LabelBinarizer()
+        y_train = lb.fit_transform(df[target])
+
+    training_df = df.drop(target, axis=1)
+    trainer = trainer_dict[mode](overwrite=True, max_trials=max_trials)
+
+    # Save the column names of all numeric columns before transforming any categorical columns into dummies
+    numeric_column_names = training_df.select_dtypes(include=[np.number]).columns.values.tolist()
+    training_df = pd.get_dummies(training_df)
+    categorical_dummy_column_names = [
+        col for col in training_df.columns.values.tolist() if col not in numeric_column_names
+    ]
+    trainer.fit(training_df, y_train, verbose=2)
+    return trainer.export_model(), categorical_dummy_column_names
+
+
+def get_prediction_df(mindsdb_df, training_df):
+    """Gets a prediction df using the df passed by the predict() method and the training df.
+
+    mindsdb_df is passed in from the .predict() method. This drops columns when users call
+    the WHERE clause in a SQL query.
+
+    training_df is the full df from training. This contains the dropped columns, which we
+    join with the rows from the mindsdb_df.
+    """
+    # Remove column that didn't exist in the training df so that we can do filtering
+    df_to_predict = mindsdb_df.copy()
+    if "__mindsdb_row_id" in df_to_predict.columns.values.tolist():
+        df_to_predict = df_to_predict.drop("__mindsdb_row_id", axis=1)
+
+    # Filter the training df based on the predict df
+    keys = list(df_to_predict.columns.values)
+    i1 = training_df.set_index(keys).index
+    i2 = df_to_predict.set_index(keys).index
+    filtered_df = training_df[i1.isin(i2)]
+
+    if filtered_df.empty:
+        # TODO: Rephrase the exception message in a more user-friendly way
+        raise Exception(
+            "The condition(s) in the WHERE clause filtered out all the data. Please refine these and try again"
+        )
+    return filtered_df
+
+
+def get_preds_from_model(df, model, target, categorical_dummy_column_names):
+    """Gets predictions from the stored AutoKeras model."""
+    cols_to_drop = ["__mindsdb_row_id", target]
+    for col in cols_to_drop:
+        if col in df.columns.values.tolist():
+            df = df.drop(col, axis=1)
+
+    # Get dummies for any categorical columns and then populate the missing ones with zeros
+    prediction_df = pd.get_dummies(df)
```

Comment:
This would indeed be an issue, which is why we also return the dummy column names through this function. 

We then deal with that possibility below in lines `get_preds_from_model()` - see lines ~90-95

```
# Get dummies for any categorical columns and then populate the missing ones with zeros
prediction_df = pd.get_dummies(df)
for col in categorical_dummy_column_names:  # exception handler for empty columns
    if col not in prediction_df.columns.values.tolist():
        prediction_df[col] = 0
```